src/main/java/com/synopsys/integration/jenkins/scan/extension/pipeline/SecurityScanStep.java

```diff
@@ -81,10 +187,44 @@ protected Execution(@Nonnull StepContext context) throws InterruptedException, I
 
         @Override
         protected Integer run() throws IOException, InterruptedException, ScannerJenkinsException {
+            Map<String, Object> pipelineParametersMap = prepareParametersMap();
+
             return ScanCommandsFactory.createPipelineCommand(listener, envVars, launcher, node, workspace)
-                .runScanner(scanParams);
+                .runScanner(pipelineParametersMap);
         }
 
     }
 
+    // This method is for mapping all the pipeline input parameters
+    private Map<String, Object> prepareParametersMap() {
```

Comment:
Then we may need to pass all the fields through method parameters. There could be around 20 fields, so if we pass all of them in the method parameters will it be fine? Initially I wanted to do that too, but passing all these large number of parameters seemed a bit odd to me. It it is fine then it can be shifted to another class.

Reply:
I was suggesting if it is possible to pass these data through the object of the class. If it's not possible then proceed this way.

Reply:
Got it. Able to do it with the help of java `this` key.